fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1640,7 +1640,8 @@ ACTOR Future<Void> postResolution(CommitBatchContext* self) {
 		MutationRef& m = pProxyCommitData->idempotencyClears[i];
 		auto& tags = pProxyCommitData->tagsForKey(m.param1);
 		self->toCommit.addTags(tags);
-		Arena arena;
+		// We already have an arena with an appropriate lifetime handy
+		Arena& arena = pProxyCommitData->idempotencyClears.arena();
```

Comment:
Is there an equivalent arena we can use for `idempotencyIdSet` (above)?

Reply:
CommitBatchContext::arena seems like a good candidate. We should probably be using this to allocate any memory that doesn't outlive the current commit batch. E.g. we could replace logRangeMutationsArena with just arena, and refactor IdempotencyIdKVBuilder to take an arena and use that. There's also (probably) a bunch of malloc calls that we could in theory convert to use this arena.

Reply:
Changed to CommitBatchContext::arena